pkg/virt-launcher/virtwrap/converter/converter.go

```diff
@@ -1973,3 +1990,48 @@ func hasTabletDevice(vmi *v1.VirtualMachineInstance) bool {
 	}
 	return false
 }
+
+func CalcDomDevice(ctx *ConverterContext, iface v1.VirtualMachineInstanceNetworkInterface) api.Interface {
+	return api.Interface{
+		Type: "ethernet",
+		Target: &api.InterfaceTarget{
+			Device:  generateTapDeviceName(iface.Name),
+			Managed: "no",
+		},
+		Model: &api.Model{Type: translateModel(ctx, "virtio")},
+		Alias: api.NewUserDefinedAlias("hotplug-" + iface.Name),
+	}
+}
+
+func generateTapDeviceName(networkName string) string {
+	const maxIfaceNameLen = 15
```

Comment:
@maiqueb  Thanks for the advice, and by the way this PR is great work, helps me a lot !
 
I did the change as your suggest, change maxIfaceNameLen from 15 to 11, it turns out two place needs to change: 
1.  one with previous comment in pkg/network/namescheme/networknamescheme.go] `MaxIfaceNameLen`
2. and `maxIfaceNameLen` in func generateTapDeviceName

if missing second change, following error occur ( event log)
```
default     0s          Normal    AddedInterface          pod/virt-launcher-vm-fedora-svx54   pod [default/virt-launcher-vm-fedora-svx54]: added interface net49969d98 to network: flannel
default     0s          Warning   SyncFailed              virtualmachineinstance/vm-fedora    server error. command SyncVMI failed: "LibvirtError(Code=1, Domain=10, Message='internal error: target managed='no' but specified dev doesn't exist')"
```

after two place change, both addinterface and removeinterface are ok now, but somehow pod's attached three interface remains, don't know why yet.

```
8: net49969d98-nic@if7: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1450 qdisc noqueue master k6t-net4996 state UP group default
    link/ether d6:36:af:0f:b3:73 brd ff:ff:ff:ff:ff:ff link-netnsid 0
9: net49969d98: <BROADCAST,NOARP> mtu 1500 qdisc noop state DOWN group default
    link/ether ba:f9:02:0c:50:70 brd ff:ff:ff:ff:ff:ff
    inet 192.168.0.26/24 brd 192.168.0.255 scope global net49969d98
       valid_lft forever preferred_lft forever
10: k6t-net4996: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1450 qdisc noqueue state UP group default
    link/ether d6:36:af:0f:b3:73 brd ff:ff:ff:ff:ff:ff
    inet 169.254.75.11/32 scope global k6t-net4996
       valid_lft forever preferred_lft forever
11: tap49969d98: <NO-CARRIER,BROADCAST,MULTICAST,UP> mtu 1450 qdisc pfifo_fast master k6t-net4996 state DOWN group default qlen 1000
    link/ether f6:bd:e0:11:80:60 brd ff:ff:ff:ff:ff:ff
``` 
event log
```
default     0s          Normal    AddedInterface          pod/virt-launcher-vm-fedora-qgqvh   Add net49969d98 [192.168.0.26/24] from cbr0
default     0s          Normal    AddedInterface          pod/virt-launcher-vm-fedora-qgqvh   pod [default/virt-launcher-vm-fedora-qgqvh]: added interface net49969d98 to network: flannel
default     0s          Warning   SyncFailed              virtualmachineinstance/vm-fedora    server error. command SyncVMI failed: "LibvirtError(Code=99, Domain=20, Message='device not found: no device found at address '(<null>)' matching MAC address '52:54:00:47:b1:bb' and alias 'ua-hotplug-flannel_net1'')"
```

Reply:
Hm, those are supposed to be there.
- net49969d98-nic is the renamed original pod interface. This will be connected to the in pod bridge.
- net49969d98 is a dummy we create so kubelet does not kill the pod over failed CNI CMD_CHECKs
- k6t-net4996 is the in pod bridge we use to extend L2 from the pod interface to the VM
- tap49969d98 is the tap device we create and configure, from which QEMU will create an emulated network interface from

If you check a random launcher pod, it will also have 4 interfaces for each network / interface pair you specify in the spec. (actually 3 if the network does **not** feature IPAM).

Reply:
Resolving this comment ...